### ISSUE TITLE
Dropdown: keyboard navigation doesn't work on dropdown with groups

### ISSUE THREAD

**Issue by hitosu** [primefaces/primereact#6124]
### Describe the bug

Keyboard navigation doesn't work on Dropdown with groups because `findNextOptionIndex` and `findPrevOptionIndex` uses top level of `visibleOptions` array to find next / prev position but in case of groups that array consists of group items and actual items are inside `items` property.

### Reproducer

https://primereact.org/dropdown/#group

### PrimeReact version

10.5.1

### React version

17.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

### Steps to reproduce the behavior

1. Open Dropdown with groups
2. Try to keyboard navigate with arrows

### Expected behavior

`findNextOptionIndex` and `findPrevOptionIndex` should look into `items` property for actual items.

**Comment by KirilCycle** [comment]
@melloware Hello! Can i take this ?

**Comment by melloware** [comment]
sure!
